### PR TITLE
fix: let danger print a compact summary of warning and error counts - WPB-9872

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9872" title="WPB-9872" target="_blank">TaskWPB-9872</a>  Show xcodebuild warnings on GitHub
  </td></table>
  <br />
 

### Issue

Currently we have too many warnings.
This PR adds another Dangerfile which is the default for all projects.
Subsequently we can revert this change project by project.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: Dangerfile.warning-workaround [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require 'json'
+
+xcode_summary.inline_mode = true
+xcode_summary.report ENV["XCRESULT_PATH"]
+
+# add extra warning
+result = xcode_summary.warning_error_count 'WireiOS.xcresult'
```
Comment (netbe) [inline Dangerfile.warning-workaround]:
```suggestion
result = xcode_summary.warning_error_count 'ENV["XCRESULT_PATH"]
```

File: Dangerfile.warning-workaround [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+max_warnings = 359
+# set current branch warnings count
+if warnings_count < max_warnings 
+    warn " You removed #{max_warnings - warnings_count} warnings !! please change max_warnings limit before merging"
+elsif warnings_count == max_warnings 
+    message "The whole team congratulates you, you did not make it worse, thanks!"
+else 
+    fail "#{warnings_count - max_warnings} warnings introduced - please fix them"
+end
```
Comment (netbe) [inline Dangerfile.warning-workaround]: so the workaround was to display a message when we reach a certain number of warning which is specify here. but this works for Wire-iOS only

File: .github/workflows/_reusable_run_tests.yml [hunk 3]
```diff
@@ -123,6 +123,13 @@ jobs:
           path: Carthage
           key: ${{ runner.os }}-xcode${{ steps.xcode-version.outputs.XCODE_VERSION }}-carthage-${{ hashFiles('Cartfile.resolved') }}
 
+      - name: Restore Danger results
+        id: restore-danger-results
+        uses: actions/cache/restore@v4
+        with:
+          path: Danger.results
+          key: ${{ runner.os }}-Danger.results
```
Comment (markusfassbender) [inline .github/workflows/_reusable_run_tests.yml]: Maybe the Xcode version should be included in the cache key, as new versions with different Swift versions can have more or less warnings?
`xcode${{ steps.xcode-version.outputs.XCODE_VERSION }}`

Comment (caldrian) [reply .github/workflows/_reusable_run_tests.yml]: done, thanks

**Review by markusfassbender — APPROVED** [review]

File: Dangerfile.warning-workaround [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+current_result = JSON.parse(warning_error_count, symbolize_names: true)
+
+directory = "Danger.results"
+filename = directory + "/" + ENV["XCRESULT_PATH"]
+Dir.mkdir(directory) unless Dir.exist?(directory)
+
+# read previous values
+previous_result = { warnings: 0, errors: 0 }
+if File.exist?(filename)
+  previous_result = JSON.parse(File.read(filename), symbolize_names: true)
```
Comment (netbe) [inline Dangerfile.warning-workaround:18]: it needs to run first on develop, no?

Comment (caldrian) [reply Dangerfile.warning-workaround:18]: Yes, but in the worst case each PR which touches a module first will show a misleading warning.

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**